### PR TITLE
Add changes to merge tempest into rjil-cicd

### PR DESCRIPTION
**Pull request by pandeyop**
Includes new repositories to be part of rjil-cicd package

The changes were tested in the gate environment. Listed below are the (partial) logs on run_tempest.sh. Some errors were removed from this log which were irrelevant.

pandeyop@gcp1-testjenkins-puppet-rjil-gate-1159:/home/jenkins/tempest$ sudo ./run_tempest.sh -N tempest.api.compute.flavors
2015-04-29 09:51:37.319 9251 WARNING oslo_config.cfg [-] Option "allow_tenant_isolation" from group "compute" is deprecated. Use option "allow_tenant_isolation" from group "auth".
2015-04-29 09:51:39.317 9258 WARNING oslo_config.cfg [-] Option "allow_tenant_isolation" from group "compute" is deprecated. Use option "allow_tenant_isolation" from group "auth".
2015-04-29 09:51:39.317 9260 WARNING oslo_config.cfg [-] Option "allow_tenant_isolation" from group "compute" is deprecated. Use option "allow_tenant_isolation" from group "auth".
2015-04-29 09:51:39.323 9262 WARNING oslo_config.cfg [-] Option "allow_tenant_isolation" from group "compute" is deprecated. Use option "allow_tenant_isolation" from group "auth".
tempest.api.compute.flavors.test_flavors_negative.FlavorDetailsNegativeTestJSON
    test_flavor_details_negative[gate,negative](inv_res_flavor)       OK  0.27
tempest.api.compute.flavors.test_flavors_negative.FlavorsListWithDetailsNegativeTestJSON
    test_flavors_list_with_details_negative[gate,negative](minDisk_gen_none)OK  0.47
    test_flavors_list_with_details_negative[gate,negative](minDisk_gen_string)OK  0.03
    test_flavors_list_with_details_negative[gate,negative](minRam_gen_none)OK  0.04
    test_flavors_list_with_details_negative[gate,negative](minRam_gen_string)OK  0.04
tempest.api.compute.flavors.test_flavors.FlavorsV2TestJSON
    test_get_flavor[gate,id-1f12046b-753d-40d2-abb6-d8eb8b30cb2f,smoke]OK  0.79
    test_list_flavors[gate,id-e36c0eaa-dff5-4082-ad1f-3f9a80aa3f59,smoke]OK  0.13
    test_list_flavors_detailed_filter_by_min_disk[gate,id-3df2743e-3034-4e57-a4cb-b6527f6eac79]OK  0.10
    test_list_flavors_detailed_filter_by_min_ram[gate,id-09fe7509-b4ee-4b34-bf8b-39532dc47292]OK  0.11
    test_list_flavors_detailed_limit_results[gate,id-b26f6327-2886-467a-82be-cef7a27709cb]OK  0.06
    test_list_flavors_detailed_using_marker[gate,id-6db2f0c0-ddee-4162-9c84-0703d3dd1107]OK  0.13
    test_list_flavors_filter_by_min_disk[gate,id-10645a4d-96f5-443f-831b-730711e11dd4]OK  0.12
    test_list_flavors_filter_by_min_ram[gate,id-935cf550-e7c8-4da6-8002-00f92d5edfaa]OK  0.10
    test_list_flavors_limit_results[gate,id-8d7691b3-6ed4-411a-abc9-2839a765adab]OK  0.05
    test_list_flavors_using_marker[gate,id-e800f879-9828-4bd0-8eae-4f17189951fb]OK  0.12
    test_list_flavors_with_detail[gate,id-6e85fde4-b3cd-4137-ab72-ed5f418e8c24,smoke]OK  0.11

Slowest 3 tests took 1.53 secs:
tempest.api.compute.flavors.test_flavors.FlavorsV2TestJSON
    test_get_flavor[gate,id-1f12046b-753d-40d2-abb6-d8eb8b30cb2f,smoke]   0.79
tempest.api.compute.flavors.test_flavors_negative.FlavorDetailsNegativeTestJSON
    test_flavor_details_negative[gate,negative](inv_res_flavor)           0.27
tempest.api.compute.flavors.test_flavors_negative.FlavorsListWithDetailsNegativeTestJSON
    test_flavors_list_with_details_negative[gate,negative](minDisk_gen_none)  0.47

Ran 22 tests in 7.349s

### CODE REVIEW

**Comment by pandeyop** [comment]
@sorenh  @rickysarraf Please review and merge.

**Comment by pandeyop** [comment]
The following packages need to be rebased in the Jiocloud umbrella.

oslo.config
oslo.utils
oslo.i18n
oslo.serializtion
msgpack-python
tempest-lib (THere were changes from Faheem that need to be rebased.)
oslo.concurrency (THis also has some additional changes required to setup.cfg. @rickysarraf Please provide the changes that was tried locally. Also concurrency needs to be rebased to master from openstack and @rickysarraf changes applied on top.)
tempest (This repo needs to completely rebase with openstack's master. I, along with the team,  will work on adding the rjil changes to tempest later)


testtools (Should, for now, be frozen at 0.9.39. @rickysarraf  Please rebase master to 0.9.39. Later when upstream changes stabilize, we can rebase it to a newer version.)

**Comment by Vaidyanath** [comment]
Tested on 1161 and seems to be working fine. @rickysarraf @sorenh Please review and merge

vaidy@gcp1-testjenkins-puppet-rjil-gate-1161:/home/jenkins/tempest$ sudo ./run_tempest.sh -N tempest.api.compute.flavors
2015-04-29 11:25:24.557 10179 WARNING oslo_config.cfg [-] Option "allow_tenant_isolation" from group "compute" is deprecated. Use option "allow_tenant_isolation" from group "auth".
2015-04-29 11:25:26.270 10185 WARNING oslo_config.cfg [-] Option "allow_tenant_isolation" from group "compute" is deprecated. Use option "allow_tenant_isolation" from group "auth".
2015-04-29 11:25:26.273 10189 WARNING oslo_config.cfg [-] Option "allow_tenant_isolation" from group "compute" is deprecated. Use option "allow_tenant_isolation" from group "auth".
2015-04-29 11:25:26.279 10187 WARNING oslo_config.cfg [-] Option "allow_tenant_isolation" from group "compute" is deprecated. Use option "allow_tenant_isolation" from group "auth".
tempest.api.compute.flavors.test_flavors.FlavorsV2TestJSON
    test_get_flavor[gate,id-1f12046b-753d-40d2-abb6-d8eb8b30cb2f,smoke]OK  0.21
    test_list_flavors[gate,id-e36c0eaa-dff5-4082-ad1f-3f9a80aa3f59,smoke]OK  0.12
    test_list_flavors_detailed_filter_by_min_disk[gate,id-3df2743e-3034-4e57-a4cb-b6527f6eac79]OK  0.11
    test_list_flavors_detailed_filter_by_min_ram[gate,id-09fe7509-b4ee-4b34-bf8b-39532dc47292]OK  0.11
    test_list_flavors_detailed_limit_results[gate,id-b26f6327-2886-467a-82be-cef7a27709cb]OK  0.05
    test_list_flavors_detailed_using_marker[gate,id-6db2f0c0-ddee-4162-9c84-0703d3dd1107]OK  0.13
    test_list_flavors_filter_by_min_disk[gate,id-10645a4d-96f5-443f-831b-730711e11dd4]OK  0.12
    test_list_flavors_filter_by_min_ram[gate,id-935cf550-e7c8-4da6-8002-00f92d5edfaa]OK  0.10
2015-04-29 11:25:29.076 10189 ERROR tempest.test [-] teardown of credentials failed: An object with that identifier already exists
Details: {u'message': u'Removing default security group not allowed.', u'type': u'ContrailConflictError', u'detail': u''}
2015-04-29 11:25:29.076 10189 TRACE tempest.test Traceback (most recent call last):
2015-04-29 11:25:29.076 10189 TRACE tempest.test   File "/home/jenkins/tempest/tempest/test.py", line 289, in tearDownClass
2015-04-29 11:25:29.076 10189 TRACE tempest.test     teardown()
2015-04-29 11:25:29.076 10189 TRACE tempest.test   File "/home/jenkins/tempest/tempest/test.py", line 407, in clear_isolated_creds
2015-04-29 11:25:29.076 10189 TRACE tempest.test     cls.isolated_creds.clear_isolated_creds()
2015-04-29 11:25:29.076 10189 TRACE tempest.test   File "/home/jenkins/tempest/tempest/common/isolated_creds.py", line 476, in clear_isolated_creds
2015-04-29 11:25:29.076 10189 TRACE tempest.test     self._cleanup_default_secgroup(creds.tenant_id)
2015-04-29 11:25:29.076 10189 TRACE tempest.test   File "/home/jenkins/tempest/tempest/common/isolated_creds.py", line 429, in _cleanup_default_secgroup
2015-04-29 11:25:29.076 10189 TRACE tempest.test     net_client.delete_security_group(secgroup['id'])
2015-04-29 11:25:29.076 10189 TRACE tempest.test   File "/home/jenkins/tempest/tempest/services/network/json/network_client.py", line 112, in _delete
2015-04-29 11:25:29.076 10189 TRACE tempest.test     resp, body = self.delete(uri)
2015-04-29 11:25:29.076 10189 TRACE tempest.test   File "/usr/lib/python2.7/dist-packages/tempest_lib/common/rest_client.py", line 283, in delete
2015-04-29 11:25:29.076 10189 TRACE tempest.test     return self.request('DELETE', url, extra_headers, headers, body)
2015-04-29 11:25:29.076 10189 TRACE tempest.test   File "/usr/lib/python2.7/dist-packages/tempest_lib/common/rest_client.py", line 629, in request
2015-04-29 11:25:29.076 10189 TRACE tempest.test     resp, resp_body)
2015-04-29 11:25:29.076 10189 TRACE tempest.test   File "/usr/lib/python2.7/dist-packages/tempest_lib/common/rest_client.py", line 685, in _error_checker
2015-04-29 11:25:29.076 10189 TRACE tempest.test     raise exceptions.Conflict(resp_body)
2015-04-29 11:25:29.076 10189 TRACE tempest.test Conflict: An object with that identifier already exists
2015-04-29 11:25:29.076 10189 TRACE tempest.test Details: {u'message': u'Removing default security group not allowed.', u'type': u'ContrailConflictError', u'detail': u''}
2015-04-29 11:25:29.076 10189 TRACE tempest.test 
    test_list_flavors_limit_results[gate,id-8d7691b3-6ed4-411a-abc9-2839a765adab]OK  0.05
    test_list_flavors_using_marker[gate,id-e800f879-9828-4bd0-8eae-4f17189951fb]OK  0.14
    test_list_flavors_with_detail[gate,id-6e85fde4-b3cd-4137-ab72-ed5f418e8c24,smoke]OK  0.11
2015-04-29 11:25:29.439 10185 ERROR tempest.test [-] teardown of credentials failed: An object with that identifier already exists
Details: {u'message': u'Removing default security group not allowed.', u'type': u'ContrailConflictError', u'detail': u''}
2015-04-29 11:25:29.439 10185 TRACE tempest.test Traceback (most recent call last):
2015-04-29 11:25:29.439 10185 TRACE tempest.test   File "/home/jenkins/tempest/tempest/test.py", line 289, in tearDownClass
2015-04-29 11:25:29.439 10185 TRACE tempest.test     teardown()
2015-04-29 11:25:29.439 10185 TRACE tempest.test   File "/home/jenkins/tempest/tempest/test.py", line 407, in clear_isolated_creds
2015-04-29 11:25:29.439 10185 TRACE tempest.test     cls.isolated_creds.clear_isolated_creds()
2015-04-29 11:25:29.439 10185 TRACE tempest.test   File "/home/jenkins/tempest/tempest/common/isolated_creds.py", line 476, in clear_isolated_creds
2015-04-29 11:25:29.439 10185 TRACE tempest.test     self._cleanup_default_secgroup(creds.tenant_id)
2015-04-29 11:25:29.439 10185 TRACE tempest.test   File "/home/jenkins/tempest/tempest/common/isolated_creds.py", line 429, in _cleanup_default_secgroup
2015-04-29 11:25:29.439 10185 TRACE tempest.test     net_client.delete_security_group(secgroup['id'])
2015-04-29 11:25:29.439 10185 TRACE tempest.test   File "/home/jenkins/tempest/tempest/services/network/json/network_client.py", line 112, in _delete
2015-04-29 11:25:29.439 10185 TRACE tempest.test     resp, body = self.delete(uri)
2015-04-29 11:25:29.439 10185 TRACE tempest.test   File "/usr/lib/python2.7/dist-packages/tempest_lib/common/rest_client.py", line 283, in delete
2015-04-29 11:25:29.439 10185 TRACE tempest.test     return self.request('DELETE', url, extra_headers, headers, body)
2015-04-29 11:25:29.439 10185 TRACE tempest.test   File "/usr/lib/python2.7/dist-packages/tempest_lib/common/rest_client.py", line 629, in request
2015-04-29 11:25:29.439 10185 TRACE tempest.test     resp, resp_body)
2015-04-29 11:25:29.439 10185 TRACE tempest.test   File "/usr/lib/python2.7/dist-packages/tempest_lib/common/rest_client.py", line 685, in _error_checker
2015-04-29 11:25:29.439 10185 TRACE tempest.test     raise exceptions.Conflict(resp_body)
2015-04-29 11:25:29.439 10185 TRACE tempest.test Conflict: An object with that identifier already exists
2015-04-29 11:25:29.439 10185 TRACE tempest.test Details: {u'message': u'Removing default security group not allowed.', u'type': u'ContrailConflictError', u'detail': u''}
2015-04-29 11:25:29.439 10185 TRACE tempest.test 
2015-04-29 11:25:29.743 10187 ERROR tempest.test [-] teardown of credentials failed: An object with that identifier already exists
Details: {u'message': u'Removing default security group not allowed.', u'type': u'ContrailConflictError', u'detail': u''}
2015-04-29 11:25:29.743 10187 TRACE tempest.test Traceback (most recent call last):
2015-04-29 11:25:29.743 10187 TRACE tempest.test   File "/home/jenkins/tempest/tempest/test.py", line 289, in tearDownClass
2015-04-29 11:25:29.743 10187 TRACE tempest.test     teardown()
2015-04-29 11:25:29.743 10187 TRACE tempest.test   File "/home/jenkins/tempest/tempest/test.py", line 407, in clear_isolated_creds
2015-04-29 11:25:29.743 10187 TRACE tempest.test     cls.isolated_creds.clear_isolated_creds()
2015-04-29 11:25:29.743 10187 TRACE tempest.test   File "/home/jenkins/tempest/tempest/common/isolated_creds.py", line 476, in clear_isolated_creds
2015-04-29 11:25:29.743 10187 TRACE tempest.test     self._cleanup_default_secgroup(creds.tenant_id)
2015-04-29 11:25:29.743 10187 TRACE tempest.test   File "/home/jenkins/tempest/tempest/common/isolated_creds.py", line 429, in _cleanup_default_secgroup
2015-04-29 11:25:29.743 10187 TRACE tempest.test     net_client.delete_security_group(secgroup['id'])
2015-04-29 11:25:29.743 10187 TRACE tempest.test   File "/home/jenkins/tempest/tempest/services/network/json/network_client.py", line 112, in _delete
2015-04-29 11:25:29.743 10187 TRACE tempest.test     resp, body = self.delete(uri)
2015-04-29 11:25:29.743 10187 TRACE tempest.test   File "/usr/lib/python2.7/dist-packages/tempest_lib/common/rest_client.py", line 283, in delete
2015-04-29 11:25:29.743 10187 TRACE tempest.test     return self.request('DELETE', url, extra_headers, headers, body)
2015-04-29 11:25:29.743 10187 TRACE tempest.test   File "/usr/lib/python2.7/dist-packages/tempest_lib/common/rest_client.py", line 629, in request
2015-04-29 11:25:29.743 10187 TRACE tempest.test     resp, resp_body)
2015-04-29 11:25:29.743 10187 TRACE tempest.test   File "/usr/lib/python2.7/dist-packages/tempest_lib/common/rest_client.py", line 685, in _error_checker
2015-04-29 11:25:29.743 10187 TRACE tempest.test     raise exceptions.Conflict(resp_body)
2015-04-29 11:25:29.743 10187 TRACE tempest.test Conflict: An object with that identifier already exists
2015-04-29 11:25:29.743 10187 TRACE tempest.test Details: {u'message': u'Removing default security group not allowed.', u'type': u'ContrailConflictError', u'detail': u''}
2015-04-29 11:25:29.743 10187 TRACE tempest.test 
tearDownClass (tempest.api.compute.flavors.test_flavors_negative
    FlavorDetailsNegativeTestJSON)                                    FAIL
    FlavorsListWithDetailsNegativeTestJSON)                           FAIL
tearDownClass (tempest.api.compute.flavors.test_flavors
    FlavorsV2TestJSON)                                                FAIL
tempest.api.compute.flavors.test_flavors_negative.FlavorsListWithDetailsNegativeTestJSON
    test_flavors_list_with_details_negative[gate,negative](minDisk_gen_none)OK  0.21
    test_flavors_list_with_details_negative[gate,negative](minDisk_gen_string)OK  0.03
    test_flavors_list_with_details_negative[gate,negative](minRam_gen_none)OK  0.03
    test_flavors_list_with_details_negative[gate,negative](minRam_gen_string)OK  0.03
tempest.api.compute.flavors.test_flavors_negative.FlavorDetailsNegativeTestJSON
    test_flavor_details_negative[gate,negative](inv_res_flavor)       OK  0.18

File: debian/control.stub [hunk 1]
```diff
@@ -2,13 +2,19 @@ Source: rjil-cicd
...
                python-pbr (>= 0.6),
                python-setuptools,
                python-sphinx,
                python-prettytable,
                python-dev,
+               python-extras,
+               python-fixtures,
+               python-mimeparse,
+               python-testresources,
+               python-twisted,
```
Comment (sorenh) [inline debian/control.stub]: Why are these needed for building?

Comment (pandeyop) [inline debian/control.stub]: These are part of the tempest dependencies. We weren't able to run tempest without these installed in the Gate Environments. Once we built the rjil-cicd package containing this and installed on the Gate environments the tempest tests worked.

Comment (sorenh) [inline debian/control.stub]: I'm not asking why they're needed at runtime. I'm asking why they're needed at package build time.

Comment (rickysarraf) [inline debian/control.stub]: Newer tempest has dependency on testtools >= 0.9.36
And most of those dependencies are for testtools.

Comment (rickysarraf) [inline debian/control.stub]: The latest testtools (1.x) have more painful dependencies. Hence, post merge, we'll rebase testtools to version > 0.9.36, but <1.x.

This is until all other dependent modules normalize.

Comment (rickysarraf) [inline debian/control.stub]: @sorenh @pandeyop Can we get this merged now ? The build-deps are a dependency because some of the setup.py files (testtools' setup.py, for example) have those modules in use.

Comment (pandeyop) [inline debian/control.stub]: @sorenh If you feel it appropriate now, kindly merge this.

**Comment by rickysarraf** [comment]
@pandeyop Until this is merged, please keep the list of repositories that we had forked/rebased. When merging this, we definitely need that detail. Otherwise the 1st build will fail.

Right now, it build fine of the local dev box because we carefully checked out the appropriate versions.

**Comment by rickysarraf** [comment]
@sorenh Can you please review this one ? This PR needs to be merged to get tempest to build.

**Comment by bodepd** [comment]
superseded by #17
